neo4j/work/simple.py

```diff
@@ -226,11 +219,41 @@ def done(summary_metadata):
             on_success=run_metadata["on_success"],
             on_failure=run_metadata["on_failure"],
         )
+
+        def on_success_done(summary_metadata):
+            result_metadata.update(summary_metadata)
+            bookmark = result_metadata.get("bookmark")
+            if bookmark:
+                self._bookmarks_in = tuple([bookmark])
+                self._bookmark_out = bookmark
+
+        def on_records_extend_records(records):
+            result._records.extend(hydrant.hydrate_records(result.keys(), records))
+
+        if cx.PROTOCOL_VERSION == Bolt3.PROTOCOL_VERSION:
+            fetch_size = -1
```

Comment:
Perhaps cleaner to ignore the fetch_size/n parameter on bolt3 (you would need to remove error check in bolt3 pull func